plugins/ipam/dhcp/daemon_test.go

```diff
@@ -0,0 +1,80 @@
+// Copyright 2015 CNI authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"os"
+	"path"
+	"testing"
+	"time"
+)
+
+func TestGetListener(t *testing.T) {
+	l, err := getListener()
+	if err != nil {
+		t.Errorf("Unable to get listener: %v", err)
+	}
+
+	// verify existence of dhcp socket
+	_, err = os.Stat(socketPath)
+	if err != nil {
+		t.Errorf("Socket does not exist %v: %v", socketPath, err)
+		return
+	}
+
+	// clean up
+	l.Close()
+}
+
+func TestSignalHandler(t *testing.T) {
+	cwd, err := os.Getwd()
+	if err != nil {
+		t.Errorf("Unable to get current working directory: %v", err)
+		return
+	}
+	var procAttr os.ProcAttr
+	proc, err := os.StartProcess(path.Join(cwd, "../../../bin/dhcp"), []string{"dhcp", "daemon"}, &procAttr)
+	if err != nil {
+		t.Errorf("Unable to start process: %v", err)
+		return
+	}
+
+	// wait for process initialization to finish
+	time.Sleep(1 * time.Second)
+
+	// verify existence of dhcp socket
+	_, err = os.Stat(socketPath)
+	if err != nil {
```

Comment:
Better to be a little more discerning when checking stat errors, e.g. for this specific error message the condition should really be `os.IsNotExist(err)` - otherwise, for `err != nil`, a more general message would be appropriate ("error checking socket existence:" or so)